code/controllers/subsystems/mapping.dm

```diff
@@ -44,37 +57,192 @@ SUBSYSTEM_DEF(mapping)
 		map_template_types[template.type] = template
 	return TRUE
 
-/datum/controller/subsystem/mapping/proc/loadEngine()
-	if(!engine_loader)
-		return // Seems this map doesn't need an engine loaded.
+/// Loads submaps at a specific point on the main map, using landmarks.
+/datum/controller/subsystem/mapping/proc/load_static_submaps(list/submap_paths)
+	log_mapload("Going to load [submap_paths.len] static PoI\s.")
+	for(var/thing in submap_paths)

```

Comment:
`for(var/obj/effect/landmark/submap_position/P as anything in submap_paths)` will give you an untyped loop without needing to cast.